go/vt/vtgate/planbuilder/operators/horizon_planning.go

```diff
@@ -195,6 +189,40 @@ func pushDownProjectionInApplyJoin(
 		src.ColumnsAST = append(src.ColumnsAST, col)
 	}
 
+	if p.TableID != nil {
+		derivedTbl, err := ctx.SemTable.TableInfoFor(*p.TableID)
+		if err != nil {
```

Comment:
`pushDownProjectionInApplyJoin` is very chunky, how about we split it in several sub-methods?

Reply:
it's just big-boned?

Reply:
Yeah a little bit, I don't have strong opinions